### DISCUSSION TITLE
Implementation issue, download size issue, login issue

### DISCUSSION THREAD

**Discussion by okieselbach** [okieselbach/Autopilot-Manager#1] [General]
changed the initial welcome discussion post from here to a problem discussion 👍

**Comment by nxtlevelj** [comment]
Hi, Do you have an updated version of the ap software? When I attempt to run it I get this version of ap.exe is not compatible.

**Reply by okieselbach** [reply]
This typically occurs when the download was not correct. curl downloads an error page like 404 file not found for example and places it into the download folder with approx. 1kb in size. When you execute this, it typically results in the "not compatible" message. So, make sure the download was really successful, more than 300kb in size.

**Reply by nxtlevelj** [reply]
Thank you. I will review the process and try it again.

**Reply by nxtlevelj** [reply]
After another attempt at it. I now get this. I’m using my Client URL and AppServiceURL which I’m guessing is the same url. I just feel like I’m missing something. 
![IMG_0143](https://github.com/user-attachments/assets/5e18f67a-f73e-4f30-9ea4-76cf81dd03cb)

**Reply by okieselbach** [reply]
Two times the app service URL ist not working. The ap.exe needs to be hosted somewhere like a storage account or you could download or directly from the GitHub account. Here an example to use the ap.exe from the GitHub account:

`curl -o ap.exe https://raw.githubusercontent.com/okieselbach/Autopilot-Manager-Client/refs/heads/master/dist/ap.exe & ap https://autopilotmanager-XYZyourInstance.azurewebsites.net/`

but it is better to host the ap.exe somewhere under your control to simplify the URL. I have it typically hosted somewhere like ap.mycompany.com directly in the root. then I can use the command:

`curl -o ap.exe ap.mycompany.com & ap https://autopilotmanager-XYZyourInstance.azurewebsites.net/`

You might also want to have a look at the issue #26 regarding hosting and https usage.

**Comment by nxtlevelj** [comment]
I was able to get it to work! Thank you. 

![IMG_0145](https://github.com/user-attachments/assets/f419dc36-9abd-4e5c-bd0e-a315edc1621d)

**Comment by nxtlevelj** [comment]
Ok so I know I'm just about there. Here is what I am facing now. When I scan the barcode and it takes me to sign in to my Microsoft account - when I sign in, the screen just takes me back to sign in to my account over and over. I am not getting to the Autopilot Manager screen to verify the import. 

I have all the groups created and assigned. The only thing I did change was in the Manifest I changed the groupmembershipclass to "ApplicationGroup" Not sure if that changed something in the process.

**Reply by okieselbach** [reply]
Did you also add the following groups to the "Enterprise App" as described in the blog post?

Excerpt:
To assign the three groups to the application you need to open up the Azure AD > Enterprise Applications > search your Enterprise App > Users and groups > add all 3 security groups:

sec – Autopilot Manager – General Access
sec – Autopilot Manager – Viewer
sec – Autopilot Manager – Approver

**Reply by nxtlevelj** [reply]
I did.

![image](https://github.com/user-attachments/assets/08d8da6d-eb18-44f8-86c0-22c216973593)

**Reply by okieselbach** [reply]
Did you configure the https://oliverkieselbach.com/2020/12/08/autopilot-manager/#Deployment

**Step 4. Configuring App Registration authentication settings**

correctly and added the token configuration groups claim?

**Reply by nxtlevelj** [reply]
Yes I followed all steps. Here is the output. 

![image](https://github.com/user-attachments/assets/66bda471-a560-49c6-9c6b-1e93eb76ef74)

**Reply by okieselbach** [reply]
What is the red screen telling you?
Do you see the request coming in at the approval site?

**Reply by nxtlevelj** [reply]
It says, "something went wrong, we couldn't import your device. Contact the IT admin to troubleshoot. 

As for the request, where is the approval site? Maybe I missed that step?

**Comment by nxtlevelj** [comment]
Hi Okieselbach, would you have any idea on what could be causing me not being able to get to autopilot import page when signing into my account?
